programmers-coding-test/sangjun/Implement/왕실의나이트.js

```diff
@@ -0,0 +1,21 @@
+const rl = require("readline-sync");
+
+const currentLocation = rl.question("나이트의 현재 위치 입력 :\n>> ").toUpperCase().split("");
+
+const steps = [[-2, -1], [-1, -2], [1, -2], [2, -1], [2, 1], [1, 2], [-1, 2], [-2, 1]];
+const rows = ["A", "B", "C", "D", "E", "F", "G", "H"];
+const columns = ["1", "2", "3", "4", "5", "6", "7", "8"];
+
+let count, nextColumn, nextRow;
+count = nextColumn = nextRow = 0;
+
+for (let i of steps) {
+    nextRow = rows.indexOf(currentLocation[0]) + 1 + i[0];
+    nextColumn = columns.indexOf(currentLocation[1]) + 1 + i[1];
```

Comment:
i는 인덱스!!
인덱스에다가 배열처럼 접근하길래 뭔가 했네

배열을 복수형으로 “s”를 붙여서 사용하는 이유가 반복문으로 원소 빼올 때 새로운 변수명을 단수로 만들 때 알아보기 쉬워서임.
steps -> step, students -> student.. 이해가 됐기를.. 상준아 형 너무 피곤하다... 살려줘...
```suggestion
for (let step of steps) {
    nextRow = rows.indexOf(currentLocation[0]) + 1 + step[0];
    nextColumn = columns.indexOf(currentLocation[1]) + 1 + step[1];
```